src/dbnode/client/session.go

```
@@ -3232,20 +3256,32 @@ type streamBlocksResult struct {
 }
 
 func newStreamBlocksResult(
+	nCtx namespace.Context,
 	opts Options,
 	resultOpts result.Options,
 	outputCh chan<- peerBlocksDatapoint,
 	tagDecoder serialize.TagDecoder,
 	idPool ident.Pool,
 ) *streamBlocksResult {
 	return &streamBlocksResult{
-		baseBlocksResult: newBaseBlocksResult(opts, resultOpts),
+		baseBlocksResult: newBaseBlocksResult(nCtx, opts, resultOpts),
 		outputCh:         outputCh,
 		tagDecoder:       tagDecoder,
 		idPool:           idPool,
 	}
 }
 
+func getContextFor(id ident.ID, opts Options) namespace.Context {
+	schemaReg := opts.SchemaRegistry()
+	if id == nil || schemaReg == nil {
+		return namespace.Context{}
+	}
+	schema, _ := schemaReg.GetLatestSchema(id)
```

Need to propagate this error up to the caller